relay-server/src/services/store.rs

```diff
@@ -64,7 +66,9 @@ impl Producer {
             **t != KafkaTopic::Outcomes && **t != KafkaTopic::OutcomesBilling
         }) {
             let kafka_config = &config.kafka_config(*topic)?;
-            client_builder = client_builder.add_kafka_topic_config(*topic, kafka_config)?;
+            client_builder = client_builder
+                .add_kafka_topic_config(*topic, kafka_config, config.validate_kafka_topics())
+                .context(ServiceError::Kafka)?;
```

Comment:
Did you mean `map_err` here?

```suggestion
                .map_err(ServiceError::Kafka)?;
```